src/libstore/binary-cache-store.cc

```diff
@@ -449,15 +449,16 @@ std::optional<const Realisation> BinaryCacheStore::queryRealisation(const DrvOut
     auto rawOutputInfo = getFile(outputInfoFilePath);
 
     if (rawOutputInfo) {
-        return { Realisation::parse(*rawOutputInfo, outputInfoFilePath) };
+        return {Realisation::fromJSON(
+            nlohmann::json::parse(*rawOutputInfo), outputInfoFilePath)};
     } else {
         return std::nullopt;
     }
 }
 
 void BinaryCacheStore::registerDrvOutput(const Realisation& info) {
     auto filePath = realisationsPrefix + "/" + info.id.to_string() + ".doi";
-    upsertFile(filePath, info.to_string(), "text/x-nix-derivertopath");
+    upsertFile(filePath, info.toJSON(), "application/json");
```

Comment:
is the out of toJSON reproducible?

Reply:
Good question… I guess the answer is “no, but we don't really care here”:

- No because:
    1. It probably won't be reproducible across Nix versions because we might want to add new fields or extend existing fields (like we already occasionally do for the narinfo files)
    2. `nlohmann::json` (the library we're using behind the scenes to convert to json) doesn't explicitely guarantees that its output is reproducible between versions − although it seems to hold in practice. I've [asked about it](https://github.com/nlohmann/json/discussions/2526) though because it could be needed in other places
- But we don't care because these only have to be semantically equivalent (we never try to content-address them or whatever, and we want to give us the possibility of extending their format in the future anyways)

Reply:
If we start using more JSON outputs, it would be nice if the fields were alphabetically sorted by default.

Reply:
If it's just that, it's already the case − and even if there's no guaranty that the default won't change, that bit of the library is pluggable − so I'm not too worried.

The bits that might change is stuff like the representation of numbers (`1000` vs `1E3` for example). That bit is hardcoded and the author didn't give any formal guaranty that it wouldn't change in the future.